### PR TITLE
[#600] Integrate Coverage Test

### LINKED ISSUE #600: Add coverage test for RepoSense
As title. It is import for our project to have some kind of coverage test to enhance code quality.

### PR DESCRIPTION
**Pull request by fzdy1914**
Fixes #600 

We do not have a Coverage Test because our coverage was low and 
had quite a few architecture changes in the past.

Having a coverage test allows us to improve our testing with higher 
revenue and lower costs.

Let's integrate Coverage Test using Gradle Jacoco plugin and codecov.io.

Meanwhile, build.gradle do not have a consistent line indentation and a 
consistent grouping of tasks and configures. Let's make all the lines to 
have an indentation of 4 and rearrange the tasks so they have a better
grouping.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -129,4 +145,30 @@ tasks.withType(Copy) {
...
+        })
+    }
+
+    reports {
+        html.enabled = true
+        xml.enabled = true
+    }
+}
+
+defaultTasks 'clean', 'build', 'test', 'systemtest', 'frontendTest', 'coverage'
```
Comment (yamidark) [inline build.gradle]: `build` will run the `test` already, don't need to add it here

File: build.gradle [hunk 2]
```diff
@@ -6,6 +6,7 @@ plugins {
     id 'com.github.johnrengelman.shadow' version '2.0.4'
     id 'com.liferay.node' version '4.4.0'
     id 'com.github.psxpaul.execfork' version '0.1.8'
+    id 'jacoco'
```
Comment (eugenepeh) [inline build.gradle]: perhaps group all the core plugins together followed by the community plugins.

File: build.gradle [hunk 3]
```diff
@@ -15,12 +16,16 @@ node.nodeVersion = '10.5.0'
 sourceCompatibility = JavaVersion.VERSION_1_8
 targetCompatibility = JavaVersion.VERSION_1_8
 
+jacoco {
+    toolVersion = "0.8.3"
```
Comment (eugenepeh) [inline build.gradle]: same here, place the plugins configuration together, according to the arrangement at the top.

**Comment by eugenepeh** [comment]
Nits to the last paragraph 

```
Meanwhile, build.gradle do not have a consistent line indentation and a
consistent grouping of tasks and configures. Let's make all the lines to
have an indentation of 4 and rearrange the tasks so they have a better
grouping.
```

to

```
As this adds more tasks and configurations to already disordered
build.gradle file, let's also reorganize the file to have a consistent
line indentation of 4 spaces and systematic structure. 
```

**Pull request merged**